manager/allocator/cnmallocator/portallocator.go

```diff
@@ -117,16 +118,30 @@ func newPortAllocator() (*portAllocator, error) {
 	return &portAllocator{portSpaces: portSpaces}, nil
 }
 
-func newPortSpace(protocol api.PortConfig_Protocol) (*portSpace, error) {
-	masterName := fmt.Sprintf("%s-master-ports", protocol)
-	dynamicName := fmt.Sprintf("%s-dynamic-ports", protocol)
+type (
+	// these are deliberately aliases, otherwise we'd only match the same type, not signature.
+	legacyIdmConstructor = func(ds datastore.DataStore, id string, start, end uint64) (*idm.Idm, error)
+	idmConstructor       = func(id string, start, end uint64) (*idm.Idm, error)
```

Comment:
```suggestion
	idmConstructor       = func(start, end uint64) (*idm.Idm, error)
```
Without the datastore, `id` would also be unused.